exif-bug65873: exif: integer overflow in the IFD tag value offset check (bug #65873).
Upstream https://github.com/php/php-src/commit/cbcf6e1880e3

diff --git a/ext/exif/exif.c b/ext/exif/exif.c
index 9fe1c0d..998ed8e 100644
--- a/ext/exif/exif.c
+++ b/ext/exif/exif.c
@@ -2742,7 +2742,12 @@ static int exif_process_IFD_TAG(image_info_type *ImageInfo, char *dir_entry, cha
 		offset_val = php_ifd_get32u(dir_entry+8, ImageInfo->motorola_intel);
 		/* If its bigger than 4 bytes, the dir entry contains an offset. */
 		value_ptr = offset_base+offset_val;
-		if (byte_count > IFDlength || offset_val > IFDlength-byte_count || value_ptr < dir_entry) {
+        /* 
+            dir_entry is ImageInfo->file.list[sn].data+2+i*12
+            offset_base is ImageInfo->file.list[sn].data-dir_offset 
+            dir_entry - offset_base is dir_offset+2+i*12
+        */
+		if (byte_count > IFDlength || offset_val > IFDlength-byte_count || value_ptr < dir_entry || offset_val < (size_t)(dir_entry-offset_base)) {
 			/*
 			// It is important to check for IMAGE_FILETYPE_TIFF
 			// JPEG does not use absolute pointers instead its pointers are relative to the start
